superset/views/core.py

```diff
@@ -2503,6 +2504,61 @@ def stop_query(self):
             pass
         return self.json_response('OK')
 
+    @has_access_api
+    @expose('/validate_sql_json/', methods=['POST', 'GET'])
+    @log_this
+    def validate_sql_json(self):
+        """Validates that arbitrary sql is acceptable for the given database.
+        Returns a list of error/warning annotations as json.
+        """
+        sql = request.form.get('sql')
+        database_id = request.form.get('database_id')
+        schema = request.form.get('schema') or None
+        template_params = json.loads(
+            request.form.get('templateParams') or '{}')
+
+        if len(template_params) > 0:
+            # TODO: factor the Database object out of template rendering
+            #       or provide it as mydb so we can render template params
+            #       without having to also persist a Query ORM object.
+            return json_error_response(
```

Comment:
I don't think we want to return 500s here. This is an unexceptional condition, it will happen frequently, right? Maybe 404?

What's the flow expected to be? Is it something like this?

1) Client requests validation...
2) Server says "nope I can't do templates yet"
3a) Client says to itself "awww man" and stops requesting validation until the query changes substantially
or...
3b) Client disregards the response and keeps querying at reasonable points? (I sort of favor 3b because it keeps the client simple and gives us a single endpoint to concern ourselves with w.r.t. scale)

Reply:
I think ideally we'd give the client a way to know ahead of time whether a selected db can do validation at all, so we don't have to try the endpoint and trap the error. I'm not entirely sure how to plumb that part, though, so the client code I have right now does 3b — it just keeps on trying pretending nobody ever told it that it'll never work.

Reply:
I agree if we have a clear path to specifically answer the question of whether it is supported then 3a would be ideal.  What about another endpoint to ask that specific question?  `GET /query-validators/presto --> 200 OK or 404 Not Found`

Reply:
I added another layer of indirection to the config object and put the "top half" of it in FEATURE_FLAGS (which are available to the client side as well).

Now the behavior there will be that if no validator is configured at all for the backend, we don't send any validation requests. If a validator is configured, but for some reason doesn't work, we switch to 3b behavior — keep trying it, believing it should be working.